### DISCUSSION TITLE
cannot access database after setting api route

### DISCUSSION THREAD

**Discussion by alperkuertuel** [neuefische/questions#703] [Web] [answered]
Hello There,

today I tried to access the data in my database via localhost:3000/api after writing the api route. To simply check if my connection works.
I created a Schema and a `connect.js` inside of the db folder. I wrote an api route for the method `get` in `pages/api/index.js`.
I simply wanted to enter the api folder in the browser to see the response of my request which should be the array with all of the art pieces objects. Sadly it shows me an empty array.

What I did so far to figure out what the problem is or can be:
When I type in "test" in to the response and enter the browser `localhost:3000/api` it shows me "test".
So I suppose that the route it self works.

```js   
import dbConnect from "@/db/connect";
import ArtPiece from "@/db/ArtPiece";

export default async function handler(request, response) {
  await dbConnect();

  if (request.method === "GET") {
    const artPieces = await ArtPiece.find();

    return response.status(200).json(artPieces);
  }
}

```

As soon as I type artPieces in the response it shows me the empty array.

Here you can find my current dev enviroment and the branch is called: "feat/database-integration":
https://github.com/alper92/capstone-project/pull/21

**Comment by F-Kirchhoff** [comment]
Can you tell me where you use useSWR to fetch the API route? And what data did you already add to your local database? Can you send me the MONGODB_URI that you saved in your .env.local (without the user / password if its included)

**Reply by alperkuertuel** [reply]
Hi,

my env.local looks like this
`MONGODB_URI=mongodb+srv://<user>:<password>@art-cluster.vrfo0sy.mongodb.net/art-gallery?retryWrites=true&w=majority`

the json i imported is inside the folder `BackUp` called `artPieces_BackUp.json`

I use useSWR inside `_app.js` like this:

```js
    <>
      <GlobalStyle />
      <SWRConfig
        value={{
          fetcher: async (...args) => {
            const response = await fetch(...args);
            if (!response.ok) {
              throw new Error(`Request with ${JSON.stringify(args)} failed.`);
            }
            return await response.json();
          },
        }}
      >
        <Component
          {...pageProps}
          artPieces={artPieces}
          artPieceToEdit={artPieceToEdit}
          filteredCategory={filteredCategory}
          fileImageUrl={fileImageUrl}
          scrollPercent={scrollPercent}
          active={active}
          handleSetFileImageUrl={handleSetFileImageUrl}
          handleArtPieceToEdit={handleArtPieceToEdit}
          handleSetArtPieces={handleSetArtPieces}
          handleSetFilteredCategory={handleSetFilteredCategory}
          handleSetScrollPercentage={handleSetScrollPercentage}
          handleSetActive={handleSetActive}
        />
      </SWRConfig>
    </>
```

**Reply by F-Kirchhoff** [reply]
But where do you actually call "useSWR" ? you on'y specify the fetcher.

**Reply by alperkuertuel** [reply]
I only imported useSWR  in `_app.js` and specified the fetcher. Because I thought, that there will be already a response  if I type in "localhost:3000/api" in the browser.

**Reply by F-Kirchhoff** [reply]
ah thats true.

**Comment by Kheirah** [comment, marked as answer]
I have a hunch. It could be due to the name of your collection. As far as I know, Mongoose casts the name to lowercase. In your case, the collection "ArtPiece" is cast to "artpiece". And that's probably not the same name in your database.
```js
const ArtPiece = mongoose.models.ArtPiece || mongoose.model("ArtPiece", artPieceSchema);
```
You will need to write it in lowercase and try again. Like this:
```js
const ArtPiece = mongoose.models.artpiece || mongoose.model("artpiece", artPieceSchema);
```

**Reply by alperkuertuel** [reply]
tried this one out as well! that was or is not the problem.

**Reply by F-Kirchhoff** [reply]
you could also manually specify the collection inside the schema: 
```js
const artPieceSchema = new Schema({ /* ... */ }, { collection: 'artpieces' });
```

**Reply by alperkuertuel** [reply]
Apologies your answer was correct:
after typing the collection name in lower-case it created me a new empty collection.
```js
const ArtPiece = mongoose.models.artpiece || mongoose.model("artpiece", artPieceSchema);
```
I copied my array inside of it and after typing  `localhost:3000/api` it now shows me my data. I did the corrections of Felix as well with manually specifying the schema. Thanks a lot you all. This caused me a lot of headach :(
